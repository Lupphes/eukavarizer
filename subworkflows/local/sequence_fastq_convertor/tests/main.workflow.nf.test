nextflow_workflow {

    name "Test Workflow SEQUENCE_FASTQ_CONVERTOR"
    script "../main.nf"
    workflow "SEQUENCE_FASTQ_CONVERTOR"
    config "subworkflows/local/sequence_fastq_convertor/tests/nextflow.config"
    tag "subworkflows"
    tag "subworkflows_local"
    tag "sequence_fastq_convertor"

    test("FASTQ.GZ input - single end - stub") {
        options '-stub'

        when {
            workflow {
                """
                input[0] = Channel.of([
                    [id: 'test', data_type: 'fastq_gz', single_end: true, sample: 'sample1', patient: 'patient1', lane: 'L001', platform: 'illumina', flowcell: 'FC1'],
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_1.fastq.gz', checkIfExists: true)
                ])
                input[1] = file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.fasta', checkIfExists: true)
                input[2] = []
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert workflow.out.tagged_collected_fastqs != null },
                { assert snapshot(workflow.out).match() }
            )
        }
    }

    test("FASTQ.GZ input - paired end - stub") {
        options '-stub'

        when {
            workflow {
                """
                input[0] = Channel.of([
                    [id: 'test', data_type: 'fastq_gz', single_end: false, sample: 'sample1', patient: 'patient1', lane: 'L001', platform: 'illumina', flowcell: 'FC1'],
                    [
                        file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_1.fastq.gz', checkIfExists: true),
                        file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_2.fastq.gz', checkIfExists: true)
                    ]
                ])
                input[1] = file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.fasta', checkIfExists: true)
                input[2] = []
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert workflow.out.tagged_collected_fastqs != null },
                { assert snapshot(workflow.out).match() }
            )
        }
    }

    test("Uncompressed FASTQ input - paired end - stub") {
        options '-stub'

        when {
            workflow {
                """
                input[0] = Channel.of([
                    [id: 'test', data_type: 'fastq', single_end: false, sample: 'sample1', patient: 'patient1', lane: 'L001', platform: 'illumina', flowcell: 'FC1'],
                    [
                        file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_1.fastq.gz', checkIfExists: true),
                        file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_2.fastq.gz', checkIfExists: true)
                    ]
                ])
                input[1] = file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.fasta', checkIfExists: true)
                input[2] = []
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert workflow.out.tagged_collected_fastqs != null }
            )
        }
    }

    test("SRA input - stub") {
        options '-stub'

        when {
            workflow {
                """
                input[0] = Channel.of([
                    [id: 'test', data_type: 'sra', sample: 'sample1', patient: 'patient1', lane: 'L001', platform: 'illumina', flowcell: 'FC1'],
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/sra/SRR13255544.tar.gz', checkIfExists: true)
                ])
                input[1] = file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.fasta', checkIfExists: true)
                input[2] = []
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert workflow.out.tagged_collected_fastqs != null },
                { assert snapshot(workflow.out).match() }
            )
        }
    }
}
