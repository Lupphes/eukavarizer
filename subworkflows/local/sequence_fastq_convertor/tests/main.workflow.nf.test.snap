{
    "SRA input - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "data_type": "sra",
                            "sample": "sample1",
                            "patient": "patient1",
                            "platform": "illumina",
                            "flowcell": "FC1",
                            "single_end": false,
                            "read_group": "\"@RG\\tID:sample1.L001\\tPU:L001\\tSM:patient1_sample1\\tLB:sample1\\tPL:illumina\"",
                            "median_bp": 0
                        },
                        [
                            "test_1.fastq.gz:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "test_2.fastq.gz:md5,d41d8cd98f00b204e9800998ecf8427e"
                        ]
                    ]
                ],
                "tagged_collected_fastqs": [
                    [
                        {
                            "id": "test",
                            "data_type": "sra",
                            "sample": "sample1",
                            "patient": "patient1",
                            "platform": "illumina",
                            "flowcell": "FC1",
                            "single_end": false,
                            "read_group": "\"@RG\\tID:sample1.L001\\tPU:L001\\tSM:patient1_sample1\\tLB:sample1\\tPL:illumina\"",
                            "median_bp": 0
                        },
                        [
                            "test_1.fastq.gz:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "test_2.fastq.gz:md5,d41d8cd98f00b204e9800998ecf8427e"
                        ]
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.3",
            "nextflow": "25.04.8"
        },
        "timestamp": "2025-10-19T02:02:12.302806991"
    },
    "FASTQ.GZ input - paired end - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "data_type": "fastq_gz",
                            "single_end": false,
                            "sample": "sample1",
                            "patient": "patient1",
                            "platform": "illumina",
                            "flowcell": "FC1",
                            "read_group": "\"@RG\\tID:FC1.sample1.L001\\tPU:L001\\tSM:patient1_sample1\\tLB:sample1\\tPL:illumina\"",
                            "median_bp": 0
                        },
                        [
                            "test_1.fastq.gz:md5,4161df271f9bfcd25d5845a1e220dbec",
                            "test_2.fastq.gz:md5,2ebae722295ea66d84075a3b042e2b42"
                        ]
                    ]
                ],
                "tagged_collected_fastqs": [
                    [
                        {
                            "id": "test",
                            "data_type": "fastq_gz",
                            "single_end": false,
                            "sample": "sample1",
                            "patient": "patient1",
                            "platform": "illumina",
                            "flowcell": "FC1",
                            "read_group": "\"@RG\\tID:FC1.sample1.L001\\tPU:L001\\tSM:patient1_sample1\\tLB:sample1\\tPL:illumina\"",
                            "median_bp": 0
                        },
                        [
                            "test_1.fastq.gz:md5,4161df271f9bfcd25d5845a1e220dbec",
                            "test_2.fastq.gz:md5,2ebae722295ea66d84075a3b042e2b42"
                        ]
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.3",
            "nextflow": "25.04.8"
        },
        "timestamp": "2025-10-20T22:58:57.412867116"
    },
    "CRAM input - stub": {
        "content": [
            {
                "0": [
                    
                ],
                "tagged_collected_fastqs": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.3",
            "nextflow": "25.04.8"
        },
        "timestamp": "2025-10-19T02:01:34.257683824"
    },
    "BAM input - stub": {
        "content": [
            {
                "0": [
                    
                ],
                "tagged_collected_fastqs": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.3",
            "nextflow": "25.04.8"
        },
        "timestamp": "2025-10-19T02:01:21.216510167"
    },
    "FASTQ.GZ input - single end - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "data_type": "fastq_gz",
                            "single_end": true,
                            "sample": "sample1",
                            "patient": "patient1",
                            "platform": "illumina",
                            "flowcell": "FC1",
                            "read_group": "\"@RG\\tID:FC1.sample1.L001\\tPU:L001\\tSM:patient1_sample1\\tLB:sample1\\tPL:illumina\"",
                            "median_bp": 0
                        },
                        "test_1.fastq.gz:md5,4161df271f9bfcd25d5845a1e220dbec"
                    ]
                ],
                "tagged_collected_fastqs": [
                    [
                        {
                            "id": "test",
                            "data_type": "fastq_gz",
                            "single_end": true,
                            "sample": "sample1",
                            "patient": "patient1",
                            "platform": "illumina",
                            "flowcell": "FC1",
                            "read_group": "\"@RG\\tID:FC1.sample1.L001\\tPU:L001\\tSM:patient1_sample1\\tLB:sample1\\tPL:illumina\"",
                            "median_bp": 0
                        },
                        "test_1.fastq.gz:md5,4161df271f9bfcd25d5845a1e220dbec"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.3",
            "nextflow": "25.04.8"
        },
        "timestamp": "2025-10-20T22:58:49.613050902"
    }
}